### ISSUE TITLE
`ios_system` should wait for all commands in line

### ISSUE THREAD

**Issue by yury** [holzschu/ios_system#73]
Example:

`echo 'nice' | ssh host`

Currently `ios_system` exits after first command (echo `nice`) and start second command (ssh host).
But we already show `blink>` prompt :)

**Comment by holzschu** [comment]
Indeed. I see where the problem comes from, but the solution is not obvious.  Commands should wait for all commands they have started (either with pipe or fork/exec/system), but not for the others. Making sure the command is not blocked by a command in another terminal is the tricky bit.

I'm going to think about it.

**Comment by yury** [comment]
I think this is related issue blinksh/blink#637

**Comment by yury** [comment]
May by whole ios_system should be as meta program...

**Comment by holzschu** [comment]
The two issues were not related, but I have committed a fix for blinksh/blink#637

I'm not sure what you mean by meta program; would you have a pointer to some documentation?

**Comment by holzschu** [comment]
Fix committed for this issue (f10015f).
It was a bug I introduced with a recent change (introduction of `pthread_detach` at line 1390 of `ios_system.m`. We've now reverted to the previous behaviour:
- wait for the first command in the pipe to terminate (with `pthread_join`) 
- then wait for the last command in the pipe to terminate (with `pthread_join` too)
Which works for these commands, and for multiple pipes too. 

The mistake I made was that once a thread has been detached, it cannot be joined again. Now, we don't detach the last command in the pipe, so we can join it later.

**Comment by yury** [comment]
Works for me. Thank you!

**Comment by yury** [comment]
`echo nice | ssh office cat` somehow try to run `office cat` as command.
ssh is using `getopts` to parse args, how can I use getopts from ios_system?

**Comment by holzschu** [comment]
Hmmm.... that's odd.

**Comment by yury** [comment]
For a test, I copied getopts from ios_system and make all vars with thread_ prefix. And it works.
